ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/VirtualArtifactRepository.java

```diff
@@ -123,8 +125,25 @@ private void transferArtifact(Object artifactModel, OutputStream destination) th
 			if (location == null) {
 				throw new FileNotFoundException(bundleInfo.getSymbolicName());
 			}
-			try (InputStream is = location.toURL().openStream()) {
-				is.transferTo(destination);
+
+			File bundleLocation = new File(location);
+			if (bundleLocation.isFile()) {
+				try (InputStream is = new FileInputStream(bundleLocation)) {
+					is.transferTo(destination);
+				}
+			} else {
+				File zipFile = null;
+				try {
+					zipFile = File.createTempFile(bundleLocation.getName(), ".jar", null); //$NON-NLS-1$
```

Comment:
Can we use a `ZipOutputStream` instead of creating a physical file to open a stream fro it and copy it to the output stream?

Reply:
Yes, that's better.  Done.

Reply:
A few lines downwards we have used a `JarOutputStream`.
Is there a reason for the difference? I wonder if just a `ZipOutputStream` would be sufficient down there too?
Looking at `JarOutputStream` the main addition to `ZipOutputStream` seems to be that it adds a `META-INF/MANIFEST.MF` entry and has a special magic-number for zips.

Reply:
It would probably better for them to be consistent.,

Reply:
`JarOutputStream` looks like a good idea, even though I think this is just jared to be extracted right away anyways but consistency is probably key here :-)